packages/flutter/test/cupertino/nav_bar_test.dart

```diff
@@ -334,6 +334,149 @@ void main() {
     expect(tester.getBottomLeft(find.text('Title')).dy, 44.0 - 8.0); // Extension gone, (static part - padding) left.
   });
 
+  testWidgets(
+      'Border is displayed by default in sliver nav bar',
+      (WidgetTester tester) async {
+    final ScrollController scrollController = new ScrollController();
+    final Key leadingKey = new GlobalKey();
+    final Key middleKey = new GlobalKey();
+    final Key trailingKey = new GlobalKey();
+    final Key titleKey = new GlobalKey();
+    await tester.pumpWidget(
+      new WidgetsApp(
+        color: const Color(0xFFFFFFFF),
+        onGenerateRoute: (RouteSettings settings) {
+          return new CupertinoPageRoute<void>(
+            settings: settings,
+            builder: (BuildContext context) {
+              return new CupertinoPageScaffold(
+                child: new CustomScrollView(
+                  controller: scrollController,
+                  slivers: <Widget>[
+                    new CupertinoSliverNavigationBar(
+                      leading: new Placeholder(key: leadingKey),
+                      middle: new Placeholder(key: middleKey),
+                      largeTitle: new Text('Large Title', key: titleKey),
+                      trailing: new Placeholder(key: trailingKey),
+                    ),
+                  ],
+                ),
+              );
+            },
+          );
+        },
+      ),
+    );
+
+    final DecoratedBox decoratedBox =
```

Comment:
Ah, sorry, the existing test wasn't ideal actually since it's not very explicit. 

Replace with 
```dart
tester.widget(find.descendant(
  of: find.byType(CupertinoSliverNavigationBar),
  matching: find.byType(DecoratedBox),
));
```
instead which says get the first DecoratedBox under CupertinoSliverNavigationBar instead of get the third DecoratedBox anywhere in the whole app.